### DISCUSSION TITLE
Is there any way to count distinct records from the API?

### DISCUSSION THREAD

**Discussion by DigNZ** [directus/directus#15485] [Archive]
I have millions of records and want to display them (a filtered subset, say around 300,000 records per query) as a heatmap - I can pull them all and map them client side but it takes a while to process and is a lot of data to send over the network. As there are so many duplicates it would be better to have a list of distinct values and a count for each as a single query.

Does this already exist in the API?
If not, would a custom endpoint be a good option?
Or reprocess the data into a separate table and keep it in sync with hooks?

Thanks.

**Comment by rijkvanzanten** [comment]
I _think_ you might get what you're looking for by grouping on the value, and then counting on that group as follows:

```
GET /items/records
  ?aggregate[count]=*
  &groupBy=field_you_want_uniques_from
```

Alternatively, a custom endpoint would be a great idea, though your third suggestion would be more performant (as you wouldn't have to re-query the full set every time)

**Reply by DigNZ** [reply]
Awesome thanks, I'll try each of those in that order and get it working.
